File: 2887-sort-vowels-in-a-string/sort-vowels-in-a-string.cpp
class Solution {
public:
    string sortVowels(string s) {
        vector<char>vowels;

        for (auto c:s){
            char ch=c;
            if (ch>=65 && ch<=90){
                ch=ch+32;
            }
            if (ch=='a' || ch=='e' || ch=='i' || ch=='o' || ch=='u'){
                vowels.push_back(c);
            }
        }

        sort(vowels.begin(),vowels.end());

        string t="";
        int i=0;

        for (auto c:s){
            char ch=c;
            if (ch>=65 && ch<=90){
                ch=ch+32;
            }
             if (ch=='a' || ch=='e' || ch=='i' || ch=='o' || ch=='u'){
                t.push_back(vowels[i]);
                i++;
            }
            else{
                t.push_back(c);
            }
            
        }
        return t;
    }
};

File: 2887-sort-vowels-in-a-string/sort-vowels-in-a-string.java
class Solution {
    public boolean isVowel(char c){
        c=Character.toLowerCase(c);
        return (c=='a' || c=='e' || c=='i' || c=='o' || c=='u' );
          
    }
    public String sortVowels(String s) {
        List<Character>vowels=new ArrayList<>();

        for (char c:s.toCharArray()){
            if (isVowel(c)) vowels.add(c);
        }

        Collections.sort(vowels);
        int ind=0;
        StringBuilder result=new StringBuilder();

        for (char c:s.toCharArray()){
            if (isVowel(c)) result.append(vowels.get(ind++));
            else result.append(c);
        }
        return result.toString();
    }
}